src/vs/workbench/api/node/extHostApiCommands.ts

```diff
@@ -258,6 +258,13 @@ export class ExtHostApiCommands {
 				{ name: 'layout', description: 'The editor layout to set.', constraint: (value: EditorGroupLayout) => typeof value === 'object' && Array.isArray(value.groups) }
 			]
 		});
+
+		this._register(OpenIssueReporter.ID, adjustHandler(OpenIssueReporter.execute), {
+			description: 'Opens the issue reporter with the provided extension id as the selected source',
+			args: [
+				{ name: 'extensionId', description: 'extensionId to report an issue on', constraint: (value: any) => typeof value === 'string' }
+			]
```

Comment:
This and the change in `apiCommands.ts` isn't needed. It's an often used pattern and little known is that you can register a command with description/constraints anywhere, e.g. use `registerCommand(command: ICommand): IDisposable` add `ICommand#description` and you have it, all commands with a description will be part of the API-docs page (no matter where they are defined) and all commands can be called via `executeCommand` anyways

Reply:
This is a sample of an api-command that defined in the 'core': https://github.com/Microsoft/vscode/blob/d8b9a524941c5224259495770f7c3e64f5f24a90/src/vs/editor/contrib/referenceSearch/referenceSearch.ts#L159

Reply:
Oh, cool, that's good to know!

I moved the description over to the existing workbench command, but then ran into an issue with calling it from the Help menu. Commands run from there have a "from" argument pushed to them (https://github.com/Microsoft/vscode/blob/1bfaa2357ac3d1e042be087268013cfcfc1fe5d3/src/vs/workbench/electron-browser/window.ts#L131), so the constraint that the argument should be a string failed. As far as I can tell, none of the commands run from the menu actually need this argument to run, and there's telemetry there to log that the command was run from the menu, but I was still hesitant to remove that line. So my solution right now is to keep the `vscode.openIssueReporter` command with the description.